app/Http/Requests/StoreQuizResultRequest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class StoreQuizResultRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        $rules = [];
+        foreach ($this->course->quiz->questions as $question) {
+            $rules['questions.'.$question->id] = 'required';
+        }
+
+        return $rules;
+    }
+
+    public function messages()
+    {
+        $messages = [];
+        foreach ($this->course->quiz->questions as $question) {
+            $messages['questions.'.$question->id.'.required'] = __('You must answer this question');
+        }
+
+        return $messages;
```

Comment:
I believe you are using the function to define the rules because you want to explicitly validated each question to be present and show the error message with the corresponding UI element. However; in the case of the messages I think you can write this using the standard notation.

```suggestion
        return [
            'questions.*' =>  __('You must answer this question');
        };
```